File: test-graph-ql/src/resolvers/Aircraft.ts
const Aircraft = {
    role: async ({ roleId }: any, _: any, { roleApi }: any) => {
        try {
            const response = await roleApi.getRoleAsync(roleId);

            return response.data;
        } catch (error: any) {
            if (error.response) {
                console.log(error.response);
            } else if (error.request) {
                console.log(error.request);
            } else {
                console.log(error);
            }
        }
    }
};

export default Aircraft;

File: AircraftManagement.Data/Entities/Aircraft.cs
using System.ComponentModel.DataAnnotations;

namespace AircraftManagement.Data.Entities
{
    public class Aircraft
    {
        [Required]
        [Range(minimum: 1, maximum: int.MaxValue)]
        public int Id { get; set; }
        [Required]
        [StringLength(maximumLength: 100, MinimumLength = 2)]
        public string Model { get; set; }
        [Required]
        [StringLength(maximumLength: 100, MinimumLength = 2)]
        public string Manufacturer { get; set; }
        public string Image { get; set; }

        public int RoleId { get; set; }
        public Role Role { get; set; }
    }
}
